### ISSUE TITLE
Append breaks queries

### ISSUE THREAD

**Issue by Risord** [Drizin/DapperQueryBuilder#19]
Append mess up query parameters:
```
var cn = new SqlConnection();
var qb = cn.QueryBuilder($"{ "A" }");
qb.Append($"{ "B" }");
```
Generated SQL query is expected to be: `@p0 @p1`
Actual: `@p0 @p0` => `A A`

Issue exist at least on 1.07 - 1.09

**Comment by Drizin** [comment]
Thanks for reporting.
Can you please check with latest code? (I've just pushed)
Looks like there was an issue with regex.
When you append "@p0" (without any leading spaces) to an existing command which already uses "@p0", this second occurrence was not being correctly replaced.

**Comment by Risord** [comment]
@Drizin thanks for super fast reation and indeed issue is fixed.
 I was already trying to decode that regex but you were faster.

**Comment by Drizin** [comment]
LOL. By coincidence, I had just worked a few days ago with this regex (in a [similar project](https://github.com/Drizin/EntityFrameworkCore.RawQueryBuilder)) and I had noticed that the regex was not working fine when searching for word-boundaries.

**Comment by Risord** [comment]
Found another case when prefixing with "("
Example:
```
var queryBuilder = new Microsoft.Data.SqlClient.SqlConnection()
    .QueryBuilder($"INSERT INTO table (col)")
    .Append($"({"A"}),")
    .Append($"({"B"});");
```

Could it be possible to abandon whole format => parse => format => parse based workflow and keep SQL and parameters separately? This should make it bullet proof while also reducing string creation on generation heavy queries (altough completely secondary thing).

**Comment by Risord** [comment]
Or this seems to be more general thing that parameters need to be surrounded with space?
All these columns fail:
```
var queryBuilder = new Microsoft.Data.SqlClient.SqlConnection()
    .QueryBuilder($"INSERT INTO table (colA,colB,colC)")
    .Append($"({"A1"},{"B1"},{"C1"}),")
    .Append($"({"A2"},{"B2"},{"C2"});");
```

**Comment by Drizin** [comment]
You're right. This really needs some improvement.

**Comment by Drizin** [comment]
I mean: I could fix the regex to identify things like `(@p0)` but probably this shouldn't even be based on regex - should probably have a better data structure to isolate/identify the parameters and the outer statement. Instead of mixing them together and splitting later we should keep them isolated from the beginning.

**Comment by Drizin** [comment]
@nikotree I couldn't reproduce those results. But it should be fixed with the same fix that I'm preparing for what @Risord reported. Can you send me the test scenario, so I can reproduce and confirm if it's fixed?

**Comment by Drizin** [comment]
Turns out this feature of ["reuse existing parameters"](https://github.com/Drizin/DapperQueryBuilder/pull/8) (which I made a broken regex) was creating many issues. I've fixed the regex, but to be on the safe-side I've set this option to be disabled by default.

**Comment by Drizin** [comment]
I've published new [Nuget 1.1.0](https://www.nuget.org/packages/Dapper-QueryBuilder/1.1.0).
@nikotree @Risord @terryaney can you please check if it's working fine? Thanks!

**Comment by Risord** [comment]
@Drizin so far so good including issue posted by nikotree. 👍
